src/main/scala/com/ncredinburgh/sonar/scalastyle/ScalastyleSensor.scala

```diff
@@ -116,7 +119,5 @@ class ScalastyleSensor(
 
   // sonar claims to accept null or a non zero lines, however if it is passed
   // null it blows up at runtime complaining it was passed 0
-  private def sanitiseLineNum(maybeLine: Option[Int]) =
-    if ((maybeLine getOrElse 0) != 0) maybeLine.get
-    else 1
+  private def sanitiseLineNum(maybeLine: Option[Int]) = maybeLine getOrElse 1
```

Comment:
To avoid non-zero line numbers, should we go for something like this instead `maybeLine.filter(_ != 0).getOrElse(1)`?

Reply:
If I get it right, sometimes the value will be a Some(0), and in those cases we should return 1 too ?
Is that even possible ?

Reply:
I'm thinking in something like this:

```scala
private def sanitiseLineNum(maybeLine: Option[Int]): Int = maybeLine match {
    case Some(0) => 1
    case Some(n) => n
    case None     => 1
}
```

Maybe  create a Constant, instead of using a plain literal.
Also, a negative line number doesn't make any sense  too... so maybe this will be better

```scala
private final val DefaultLineNumber = 1
/** Ensures that a line number is valid, if not returns a default value */
private def sanitiseLineNum(maybeLine: Option[Int]): Int = maybeLine match {
    case Some(linenum) if linenum > 0 => linenum
    case _  => DefaultLineNumber
}
```

Reply:
So essentially the following is a concise version of what we're trying to do:
```scala
maybeLine.filter(_ >= 0).getOrElse(1)
```

Reply:
oh yes, I will push it right now